### ISSUE TITLE
Core dump at ethr_native_atomic64_add_return_mb

### ISSUE THREAD

**Issue by renaililith** [erlang/otp#7890]
Describe the bug
We are seeing randomly coredump after gen_server:call/3 remote process timeout.
A maybe helpful info：we periodicly erlang:garbage_collect/1 the caller process.

To Reproduce
I have not a right way to reproduce this issue consistently, but opening the bug to see if you have a theory from the corefile.

Expected behavior
No coredump during after gen_server:call/3 remote process returned..

Affected versions
We are seeing this issue with OTP 26.1.2 erts-14.1.1

Additional context
The start arguments of our project are "-K true -A30 -zdbbl 131072"
Here is the backtrace from the coredump.

```
(gdb) bt
#0  0x000055ddf2e94587 in ethr_native_atomic64_add_return_mb (incr=-1, var=0xffffffffffffffd8) at ../include/internal/x86_64/../i386/atomic.h:240
#1  ethr_atomic_add_read (val=-1, var=0xffffffffffffffd8) at ../include/internal/ethr_atomics.h:4219
#2  ethr_atomic_dec_read (var=0xffffffffffffffd8) at ../include/internal/ethr_atomics.h:4806
#3  erts_refc_dectest (min_val=0, refcp=0xffffffffffffffd8) at beam/sys.h:1031
#4  erts_bin_release (bp=0xffffffffffffffd0) at beam/erl_binary.h:508
#5  de_refc_dec (min=0, dep=0x0) at beam/erl_node_tables.c:98
#6  erts_deref_dist_entry (dep=0x0) at beam/erl_node_tables.c:120
#7  0x000055ddf2e176bd in erts_free_dist_ext_copy (edep=0x7fd113ac23e0) at beam/external.c:892
#8  0x000055ddf2de37c9 in erts_cleanup_message (mp=0x7fd113ac2388) at beam/erl_message.c:227
#9  erts_cleanup_messages (msgp=msgp@entry=0x7fd113ac2388) at beam/erl_message.c:251
#10 0x000055ddf2df364a in handle_alias_message (next_nm_sig=0x7fd1a3a2c0a0, sig=0x7fd113ac2388, c_p=0x7fd1a3a2bf30) at beam/erl_proc_sig_queue.c:5451
#11 erts_proc_sig_handle_incoming (c_p=c_p@entry=0x7fd1a3a2bf30, statep=statep@entry=0x7fd379dfacb8, redsp=redsp@entry=0x7fd379dfacbc, max_reds=100, local_only=local_only@entry=0) at beam/erl_proc_sig_queue.c:6164
#12 0x000055ddf2ca51da in erts_schedule (esdp=0x7fd37eec6f80, p=0x7fd1a3a2bf30, calls=0) at beam/erl_process.c:10187
#13 0x00007fd37c263965 in ?? ()
#14 0x0000000000000000 in ?? ()
(gdb) f 10
#10 0x000055ddf2df364a in handle_alias_message (next_nm_sig=0x7fd1a3a2c0a0, sig=0x7fd113ac2388, c_p=0x7fd1a3a2bf30) at beam/erl_proc_sig_queue.c:5451
5451	in beam/erl_proc_sig_queue.c
(gdb) etp-sig-int sig
!ENCODED-DIST-MSG @from=
```

**Comment by rickard-green** [comment]
#7890 should fix this issue

**Comment by renaililith** [comment]
> BTW #7915 is based on OTP 26.1.2, so you can test it just as is

Thanks, we will try it.

**Comment by rickard-green** [comment]
This issue has been fixed in [OTP 26.2](https://www.erlang.org/patches/otp-26.2#erts-14.2), [OTP 25.2.3.8](https://www.erlang.org/patches/otp-25.3.2.8#erts-13.2.2.5), and [OTP 24.3.4.15](https://www.erlang.org/patches/otp-24.3.4.15#erts-12.3.2.15)
